Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Asset/AssetConnection.tsx

```diff
@@ -106,7 +104,31 @@ function AssetConnectionWindow(props: { Name: string, ID: number, TypeID: number
         if (index == -1 && localAssets.length > 0)
             setSelectedAssetID(localAssets[0].ID)
     }, [localAssets])
-   
+
+    React.useEffect(() => {
+        const handle = getAssetConnections();
+        return () => { if (handle != null && handle.abort != null) handle.abort(); }
+    }, [props.ID, trigger])
+
+    React.useEffect(() => {
+        const h = getLocations();
+        return () => { if (h!= null && h.abort != null) h.abort(); }
+    }, [assetConnections])
+
+    function getLocations() {
```

Comment:
This function is only used on the useEffect, it might as well be just a part of that.